test/test_manifest_providers.py

```diff
@@ -46,9 +49,46 @@ def test_git_source():
     assert '' == package_path
     assert '<version>0.5.7</version>' in package_xml
 
-
+# mock_get_url_contents is used to mock out the '_get_url_contents' method in
+# the rosdistro.manifest_provider.github module.  Instead of going out to github
+# to get data (which can get rate-limited in Travis), it instead pulls a canned
+# example from some local files.
+def mock_get_url_contents(req):
+    import re
+    import sys
+
+    # For python3, look for the 'str' type; for Python 2, the 'unicode' type
+    if sys.version_info[0] == 3:
+        text_type = str
+    else:
+        text_type = unicode
+
+    # The urlopen() function from urllib or urllib2 takes either a string or a
+    # urllib.Request object in; determine the URL in either case.
+    if isinstance(req, text_type):
+        haystack = req
+    else:
+        haystack = req.get_full_url()
```

Comment:
I probably would have been lazy and just done all of this as a big try block where the py3 code runs first and then errors out into the py2 code. But what you have here is fine.